application/src/main/java/org/togetherjava/tjbot/features/moderation/scam/TokenAnalyse.java

```diff
@@ -0,0 +1,157 @@
+package org.togetherjava.tjbot.features.moderation.scam;
+
+import org.togetherjava.tjbot.config.ScamBlockerConfig;
+import org.togetherjava.tjbot.features.utils.StringDistances;
+
+import java.net.URI;
+import java.util.Locale;
+
+/**
+ * Analyzes a given text token. Populates various metrics regarding the token possibly being
+ * suspicious, returning back results of the token analyze.
+ *
+ * Highly configurable, using {@link ScamBlockerConfig}. Entry point to use is
+ * {@link #analyze(String, AnalyseResults)}.
+ */
+final class TokenAnalyse {
+    private final ScamBlockerConfig config;
+
+    TokenAnalyse(ScamBlockerConfig config) {
+        this.config = config;
+    }
+
+    /**
+     * Analyzes the given token about being suspicious.
+     * 
+     * @param token the token to analyze
+     * @param results metrics representing how suspicious the token is
+     */
+    void analyze(String token, AnalyseResults results) {
+        if (token.isBlank()) {
+            return;
+        }
+
+        if (!results.pingsEveryone()
+                && ("@everyone".equalsIgnoreCase(token) || "@here".equalsIgnoreCase(token))) {
+            results.markPingsEveryone();
+        }
+
+        if (!results.containsSuspiciousKeyword() && containsSuspiciousKeyword(token)) {
+            results.markContainsSuspiciousKeyword();
+        }
+
+        if (!results.containsDollarSign()
+                && (token.contains("$") || "usd".equalsIgnoreCase(token))) {
+            results.markContainsDollarSign();
+        }
+
+        if (token.startsWith("http")) {
+            analyzeUrl(token, results);
+        } else {
+            results.markNonUrlTokenFound();
+        }
+    }
+
+    private boolean containsSuspiciousKeyword(String token) {
+        String preparedToken = token.toLowerCase(Locale.US);
+
+        // Checks the token against various keywords from the config
+        // The keywords support some regex-inspired syntax
+        return config.getSuspiciousKeywords()
```

Comment:
Why arent we using actual regex here?

Reply:
I am fine with it for now because the "new syntax" is kinda just a subset of regex.